product/views.py

```diff
@@ -133,4 +135,62 @@ def get(self, request, product_id):
 def get_user_info(community):
     return User.objects.values(
                 'id', 'nick_name', 'profile_image'
-           ).get(id=community.user_id)
\ No newline at end of file
+           ).get(id=community.user_id)
+
+class MainPageView(View):
+    def get(self, request):
+        try:
+            sorting          = request.GET.get('sorting')
+            main_category_id = request.GET.get('main')
+            sub_category_id  = request.GET.get('sub')
+        
+            products = Product.objects.select_related(
+                'main_category',
+                'sub_category',
+                'creator'
+                ).prefetch_related(
+                    'product_like_user',
+                    'product_view_user'
+                    ).annotate(likecount=Count('product_like_user'))
+
+            filters = {}
+
+            if main_category_id:
+                filters['main_category__id'] = main_category_id
+
+            if sub_category_id:
+                filters['sub_category__id'] = sub_category_id
+
+            sortings   = {
+                'updated' : '-created_at',
+                'popular' : '-likecount'
+            }
+
+            if sorting in sortings:
+                products = products.order_by(sortings[sorting])
+            else:
+                return JsonResponse({'MESSAGE': 'KEY_ERROR'}, status=400)
+
+            products_list = [{
+                'created_at'  : product.created_at,
+                'id'          : product.id,
+                'title'       : product.name,
+                'thumbnail'   : product.thumbnail_image,
+                'subCategory' : product.sub_category.name,
+                'creator'     : product.creator.name,
+                'isLiked'     : True if ProductLike.objects.filter(product_id=product.id).exists() else False,
```

Comment:
product 부터 접근해서 구현해보세요.
```suggestion
                'isLiked'     : True if product.xxxx else False,
```

Reply:
접근해보았는데, 아직로그인 안한 상태이기 때문에 모두 False로 넣었고, 로그인 한 상태이면 어떡할지 아직 모르겠습니다.

Reply:
재훈님!! 로그인 한 상태에서 보여주는 products와 로그인 하지 않은 상태에서 보여주는 products가 필요하다고 하면,
엔드포인트를 분리하시면 될 것 같아요!

Reply:
네 알겠습니다!